hkmc2/shared/src/test/mlscript-compile/Stack.mls

```diff
@@ -6,7 +6,7 @@ open Predef
 // TODO
 // type Stack[A] = Stack.Cons[A] | Stack.Nil
 
-module Stack[T] with ...
+object Stack[T] with ...
```

Comment:
We most definitely want this to be a module...

Reply:
This is similar to https://github.com/hkust-taco/mlscript/pull/295#discussion_r2029700651 as we are now using `Stack` as both the module `Stack` and the type `Stack`. At `GeneratorStack.mls:8` there is a reference to `Stack` which is used as a non-module type, and it crashes because of our module method checks. 

I will change this back to module. I think the way to do it is to just remove the type signature `Stack[Int]` at `GeneratorStack.mls:8` until we implement the overloading.

Reply:
Instead of removing the type signature, please change `Stack[Int]` to `Stack.T[Int]`, where `T` is defined as
```fs
type T[A] // TODO support type/module overloading
```
in the `Stack` module.